operator/controllers/kyma_controller.go

```diff
@@ -503,3 +525,56 @@ func (r *KymaReconciler) GetModules(ctx context.Context, kyma *v1alpha1.Kyma) (p
 
 	return modules, nil
 }
+
+func (r *KymaReconciler) DeleteNotExistsModulesAndUpdateConditions(ctx context.Context,
+	status *v1alpha1.KymaStatus, moduleInfos []*v1alpha1.ModuleInfo,
+) (bool, error) {
+	var err error
+	requireUpdateCondition := false
+	conditionMap := convertToConditionMap(status)
+
+	for i := range moduleInfos {
+		moduleInfo := moduleInfos[i]
+		err = r.deleteModule(ctx, moduleInfo)
+		if client.IgnoreNotFound(err) == nil {
+			requireUpdateCondition = true
+			delete(conditionMap, moduleInfo)
+		}
+	}
+
+	status.Conditions = convertToNewConditions(conditionMap)
+
+	if client.IgnoreNotFound(err) != nil {
+		return false, fmt.Errorf("error deleting module %w", err)
+	}
+	return requireUpdateCondition, nil
+}
+
+func convertToConditionMap(status *v1alpha1.KymaStatus) map[*v1alpha1.ModuleInfo]*v1alpha1.KymaCondition {
+	conditionMap := make(map[*v1alpha1.ModuleInfo]*v1alpha1.KymaCondition)
+	for i := range status.Conditions {
+		condition := &status.Conditions[i]
+		conditionMap[&condition.ModuleInfo] = condition
+	}
+	return conditionMap
+}
+
+func convertToNewConditions(conditionMap map[*v1alpha1.ModuleInfo]*v1alpha1.KymaCondition) []v1alpha1.KymaCondition {
+	newConditions := make([]v1alpha1.KymaCondition, 0)
+	for _, condition := range conditionMap {
+		newConditions = append(newConditions, *condition)
+	}
+	return newConditions
+}
+
+func (r *KymaReconciler) deleteModule(ctx context.Context, moduleInfo *v1alpha1.ModuleInfo) error {
+	module := unstructured.Unstructured{}
+	module.SetNamespace(moduleInfo.Namespace)
+	module.SetName(moduleInfo.Name)
+	module.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   moduleInfo.GroupVersionKind.Group,
+		Version: moduleInfo.GroupVersionKind.Version,
+		Kind:    moduleInfo.GroupVersionKind.Kind,
+	})
+	return r.Delete(ctx, &module, &client.DeleteOptions{})
```

Comment:
what if the module is already deleted, will this call still succeed? Also your delete is async due to you not setting the deletion propagation to foreground, is this on purpose? If it is, I think you are missing the check that the module really has been deleted before removing it from the condition.